hs_rest_api2/tests/test_file_json_metadata.py

```diff
@@ -0,0 +1,200 @@
+    return metadata_str \
+        .replace("http://www.hydroshare.org", current_site_url()) \
+        .replace("0fdbb27857844644bacc274882601598", short_id)
+
+
+def sorting(item):
+    if isinstance(item, dict):
+        return sorted((key, sorting(values)) for key, values in item.items())
+    if isinstance(item, list):
+        return sorted(sorting(x) for x in item)
+    else:
+        return item
+
+
+def prepare_resource(self, folder):
+    prepare_resource_util(folder, self.res, self.user, self.extracted_directory, self.test_bag_path)
+
+
+class TestFileBasedJSON(HSRESTTestCase):
+
+    base_dir = 'hs_rest_api2/tests/data/json/'
+
+    def __init__(self, methodName, param1=None, param2=None):
+        super(TestFileBasedJSON, self).__init__(methodName)
+
+        self.param1 = param1
+        self.param2 = param2
+
+    def setUp(self):
+        super(TestFileBasedJSON, self).setUp()
+
+        self.tmp_dir = tempfile.mkdtemp()
+
+        # create empty resource
+        self.res = resource.create_resource(
+            'CompositeResource',
+            self.user,
+            'triceratops'
+            )
+
+        self.test_bag_path = 'hs_rest_api2/tests/data/test_resource_metadata_files.zip'
+
+        self.extracted_directory = 'hs_core/tests/data/test_resource_metadata_files/'
+
+    def tearDown(self):
+        super(TestFileBasedJSON, self).tearDown()
+        os.remove(self.test_bag_path)
+        self.res.delete()
+
+    def _test_metadata_update_retrieve(self, endpoint, schema_in, json_put_file, aggregation_path=None):
+        kwargs = {"pk": self.res.short_id}
+        if aggregation_path:
+            kwargs["aggregation_path"] = aggregation_path
+        response = self.client.get(reverse(endpoint, kwargs=kwargs))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        with open(os.path.join(self.base_dir, json_put_file), "r") as f:
+            full_resource_json = json.loads(normalize_metadata(f.read(), self.res.short_id))
+        schema_in_instance = schema_in(**full_resource_json)
+        in_json = schema_in_instance.dict(exclude_defaults=True)
+        self.client.put(reverse(endpoint, kwargs=kwargs),
+                        data=in_json, format="json")
+
+        response = self.client.get(reverse(endpoint, kwargs=kwargs))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+
+        if aggregation_path:
+            # TODO remove rights from aggregation in schema
+            full_resource_json['rights'] = response_json['rights']
+        else:
+            self.assertGreater(response_json['modified'], full_resource_json['modified'])
+            # overwrite system metadata fields for comparison
+            full_resource_json['modified'] = response_json['modified']
+            full_resource_json['created'] = response_json['created']
```

Comment:
Because the created date in the test json is different than the created date for the newly created test resource.  I could remove created from the comparison, but this results to the same thing.

Reply:
When I removed this override, test did not fail.